cmd/beyla/main.go

```diff
@@ -34,6 +34,11 @@ func main() {
 		os.Exit(-1)
 	}
 
+	if err := beyla.CheckOSCapabilities(); err != nil {
```

Comment:
Having this here instead of within `RunBeyla` means we fail early and swiftly, so that we don't spend time bootstrapping something that is doomed to failure because of lacking permissions.